generated/register_all_services.cpp

```diff
@@ -0,0 +1,114 @@
+//---------------------------------------------------------------------
+// This file is automatically generated. All manual edits will be lost.
+//---------------------------------------------------------------------
+// Register all services implementation.
+//---------------------------------------------------------------------
+#include "register_all_services.h"
+
+#include <string>
+#include <vector>
+
+#include <server/core_service_registrar.h>
+#include <server/session_repository.h>
+
+#include "nidaqmx/nidaqmx_service_registrar.h"
+#include "nidcpower/nidcpower_service_registrar.h"
+#include "nidigitalpattern/nidigitalpattern_service_registrar.h"
+#include "nidmm/nidmm_service_registrar.h"
+#include "nifgen/nifgen_service_registrar.h"
+#include "nirfsa/nirfsa_service_registrar.h"
+#include "nirfsg/nirfsg_service_registrar.h"
+#include "niscope/niscope_service_registrar.h"
+#include "niswitch/niswitch_service_registrar.h"
+#include "nisync/nisync_service_registrar.h"
+#include "nitclk/nitclk_service_registrar.h"
+
+namespace nidevice_grpc {
+
+std::shared_ptr<void> register_all_services(
+  grpc::ServerBuilder& server_builder,
+  const nidevice_grpc::FeatureToggles& feature_toggles,
+  nidevice_grpc::FeatureToggles::CodeReadiness code_readiness)
+{
+  auto service_vector = std::make_shared<std::vector<std::shared_ptr<void>>>();
+
+  auto session_repository = std::make_shared<nidevice_grpc::SessionRepository>();
+  auto core_service = nidevice_grpc::register_core_service(server_builder, session_repository);
+  service_vector->insert(
+    service_vector->end(), 
+    {session_repository, core_service});
```

Comment:
Note: adding the `session_repository` to the `vector` is kind of a hack. `std::vector` doesn't guarantee any kind of destructor ordering on elements, so it could get destroyed before the refers. I started another change to always use a `shared_ptr` for that class so that it can be jointly-owned by the services. That touches a lot of files, so I'm holding that back as a separate change.

We don't really have a clean shutdown procedure other than needing it to work in tests where we've already gone into an idle state. If we were shutting down a live server, this would not be the only potential ordering issue.